pytorch_lightning/core/lightning.py

```diff
@@ -373,14 +381,23 @@ def all_gather(self, tensor: Union[torch.Tensor], group: Optional[Any] = None, s
         distributed processes
 
         Args:
-            tensor: tensor of shape (batch, ...)
+            tensor: int, float, tensor of shape (batch, ...),  or a collection of
+                int, float, tensor of shape (batch, ...)
```

Comment:
```suggestion
            tensor: int, float, tensor of shape (batch, ...), or a (possibly nested) collection thereof.
```
it can be nested arbitrarily, right?